install-devnet-packages.sh

```diff
@@ -0,0 +1,530 @@
+TOTAL_STEPS=10
+
+# Detect Operating System
+OS_TYPE=$(uname)
+
+# MacOS
+if [ "$OS_TYPE" = "Darwin" ]; then
+    echo "🚀 Starting package installation for MacOS!"
+
+# Linux
+elif [ "$OS_TYPE" = "Linux" ]; then
+
+    # Detect the Linux distribution (Ubuntu, Fedora, Arch, etc)
+    if [ -f /etc/os-release ]; then
+        . /etc/os-release
+        OS_NAME=$NAME
+    fi
+    echo "🚀 Starting package installation for Linux/$OS_NAME!"
+
+# Other OS (Windows, BSD, etc.)
+else
+    echo "$OS_TYPE is an unsupported operating system."
+fi
+
+echo
+
+# Check Shell
+if [ -n "$ZSH_VERSION" ]; then
+    SHELL_NAME="zsh"
+    echo "The current shell is $SHELL_NAME. The installation will proceed based on $SHELL_NAME."
+elif [ -n "$BASH_VERSION" ]; then
+    SHELL_NAME="bash"
+    echo "The current shell is $SHELL_NAME. The installation will proceed based on $SHELL_NAME."
+else
+    echo "The current shell is $SHELL_NAME. $SHELL_NAME is an unsupported shell."
+fi
```

Comment:
This script run in /bin/sh (without shabang, will run /bin/sh).
so, if you want user's shell env, you can use $SHELL.

Reply:
I was assuming that the installation would be based on the shell in which the user is currently entering commands. (e.g. Even if the default shell is zsh, I want the installation to proceed with bash if the file is opened in bash.) 
That's why I removed the shebang and didn't use the $SHELL. (use a command like `source install-devnet-packages.sh` to execute the script in the current shell environment.)

Reply:
Did you mean this script is intended user must run script with sourcing?

Reply:
Yes, that's correct. The script must be run using a command like sourcing that works with the current shell environment.
Or should i consider the alternative you mentioned, by using shebangs to create separate files for zsh and bash?

Reply:
Is it because this script is intended to handle everything so that users don’t need to source the installed environment themselves? Or is there another reason?

Reply:
Yes, that's correct. What I wanted was to allow the entire process, from installation to running the devnet, to be done in a single shell environment. 
That's why I used source each time the environment variables changed.

Reply:
It seems extremely rare to open and use a shell other than the default one mentioned above.
 
How about using a shebang to ensure consistent execution of the script, and after the installation is complete, printing a message instructing the user to source their respective shell environment?

Reply:
Got it. I will use shebang to seperate files for zsh and bash.

Reply:
zsh, bash 각각 파일을 분리 하라는 의미는 아니었습니다.  위에 언급한 기본쉘이 아닌 다른 쉘을 열고 인스톨 스크립트를 통해 설치하는 경우는 일반적인 상황이 아닌 극히 드문 경우인 것 같으니 이는 배제해도 좋을 것 같습니다. 그러므로 $SHELL 환경변수를 통해 사용자가 사용하고 있는 쉘 환경을 받아와도 일반적인 개발 환경에서는 충분하다고 생각합니다.

Reply:
zsh, bash 각각 파일을 분리 하라는 의미는 아니었습니다.  위에 언급한 기본쉘이 아닌 다른 쉘을 열고 인스톨 스크립트를 통해 설치하는 경우는 일반적인 상황이 아닌 극히 드문 경우인 것 같으니 이는 배제해도 좋을 것 같습니다. 그러므로 $SHELL 환경변수를 통해 사용자가 사용하고 있는 쉘 환경을 받아와도 일반적인 개발 환경에서는 충분하다고 생각합니다.

그래서 다음과 같은 제안을 한것입니다.
1. shebang 을 사용해서 이 스크립트가 일관되게 실행될 수 있도록 한다.(일반적으로 bash를 사용)
[option 1]
2. $SHELL 로 현재 사용자의 쉘의 정보를 받아와 프로파일에 경로 추가 (zshrc, bashrc, profile etc..)
3. install이 모두 완료 되고 난 후 복사/붙여넣기가 용이하게 source 할 수 있도록 안내문 출력
```bash
Finished install..
Please source your profile.
source ~/.zshrc
```

[option 2]
2. 인스톨한 도구의 경로를 추가할 필요가 있을 경우 모두 수동으로 할 수 있도록 안내문 출력
```bash
Finished install...
Please proceed with the following steps.
echo 'export PATH="$PATH:/usr/local/go/bin" >> ~/.zshrc
source ~/.zshrc
```

Reply:
넵 이해했습니다. 친절한 설명 감사합니다!

Reply:
이건 제안이고 꼭 이렇게 변경하라는 코멘트는 아닙니다! @swywssaid님이 원래 작성한 코드가 더 유용하고 좋을 수도 있으니 고민해보시고 적용하면 좋을 것 같습니다 🙇‍♂️

Reply:
넵 알겠습니다. 감사합니다! 🙇

Reply:
I applied it to commit [8d54f32](https://github.com/tokamak-network/tokamak-thanos/pull/265/commits/8d54f32cd65d05ae57f36f48c515c017b49555f9). Thank you.